python-main.js

```diff
@@ -1044,58 +1048,57 @@ function web_editor(config) {
     }
 
     function doConnect(e, serial) {
-        navigator.usb.requestDevice({
-            filters: [{vendorId: 0x0d28, productId: 0x0204}]
-        }).then(function(device) {
-            // Connect to device
-            window.transport = new DAPjs.WebUSB(device);
-            window.daplink = new DAPjs.DAPLink(window.transport);
-
-            // Ensure disconnected
-            window.daplink.disconnect();
-
-            // Connect to board
-            return window.daplink.connect()
-            .then(function() {
-                try {
-                 var output = generateFullHexStr();
-               } catch(err) {
-                 alert(err.message);
-                 return;
-                }
+        var p = Promise.resolve();
 
-                // Change button to disconnect
-                $("#command-connect").attr("id", "command-disconnect");
-                $("#command-disconnect > .roundlabel").text(config["translate"]["static-strings"]["buttons"]["command-disconnect"]["label"]);
-                $("#command-disconnect").attr("title", config["translate"]["static-strings"]["buttons"]["command-disconnect"]["title"]);
+        if (usePartialFlashing) {
+            p = PartialFlashing.connectDapAsync();
+        }
+        else {
+            p = navigator.usb.requestDevice({
+                filters: [{vendorId: 0x0d28, productId: 0x0204}]
+            }).then(function(device) {
+                // Connect to device
+                window.transport = new DAPjs.WebUSB(device);
+                window.daplink = new DAPjs.DAPLink(window.transport);
```

Comment:
If partial flashing is enabled then `window.daplink` will never be created and there is at least one call to this in that will result in a reference error:

https://github.com/bbcmicrobit/PythonEditor/blob/12101eae3148d9b6631cdfb6e4db2fec3b4ade2e/python-main.js#L1383

There might be more, so have a look around just in case.

Reply:
Don't worry about this one, the fix will be included as part of https://github.com/bbcmicrobit/PythonEditor/pull/211/files#diff-6d9d88a8fbf6c28991df9328acc4142aR1384.